### ISSUE TITLE
GraphViz: RuntimeException: Tree has not been constructed

### ISSUE THREAD

**Issue by Anecho** [andrewdalpino/Rubix-engine#249]
### Setup:
* Docker version 20.10.17, build 100c701 on WSL2
* php:7.4-fpm with all recommended and optional Rubix dependencies installed, except [Extras Package](https://github.com/RubixML/Extras)
* Laravel v8.83.23

### Example code from https://docs.rubixml.com/2.0/regressors/regression-tree.html
```php
use Rubix\ML\Regressors\RegressionTree;
use Rubix\ML\Helpers\Graphviz;
use Rubix\ML\Persisters\Filesystem;

$estimator = new RegressionTree(20, 2, 1e-3, 10, null);

$dot = $estimator->exportGraphviz();

Graphviz::dotToImage($dot)->saveTo(new Filesystem('tree.png'));
```
### Thrown RuntimeException stack: https://flareapp.io/share/LPl1A227#F44
![image](https://user-images.githubusercontent.com/2618636/191997121-a28d95d7-7a66-4867-abd2-25f195b06724.png)  
![image](https://user-images.githubusercontent.com/2618636/191997208-c45343b3-48b6-4c7f-b006-10ff76797ca7.png)  
![image](https://user-images.githubusercontent.com/2618636/191997333-75fd7eb4-3dc2-4ec2-9d76-f106971194f1.png)  

### What am I missing?

**Comment by andrewdalpino** [comment]
Hey @Anecho, it looks as though you are trying to export the graph of a tree that has not been constructed yet. To construct the tree, call the `train()` method on the Regression Tree with a training dataset object and *then* call `exportGraphviz()`.

**Comment by andrewdalpino** [comment]
@Anecho and @RubixML/engineering, I think we should maybe say something like "Tree has not been constructed, must train learner." or something like that. Users might not understand that training is equivalent to constructing a decision tree under the hood. Thoughts?

**Comment by andrewdalpino** [comment]
A more actionable error message was implemented in 2.2
